src/cdk/a11y/focus-trap/event-listener-inert-strategy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusTrapInertStrategy} from './focus-trap-inert-strategy';
+import {ConfigurableFocusTrap} from './configurable-focus-trap';
+
+/**
+ * Lightweight FocusTrapInertStrategy that adds a document focus event
+ * listener to redirect focus back inside the FocusTrap.
+ */
+export class EventListenerFocusTrapInertStrategy implements FocusTrapInertStrategy {
+  /** Focus event handler. */
+  private _listener: any;
+
+  /** Adds a document event listener that keeps focus inside the FocusTrap. */
+  preventFocus(focusTrap: ConfigurableFocusTrap): void {
+    this._listener = this._trapFocus.bind(null, focusTrap);
+    focusTrap.ngZone.runOutsideAngular(() => {
+      focusTrap.document.addEventListener('focus', this._listener, true);
+    });
+  }
+
+  /** Removes the event listener added in preventFocus. */
+  allowFocus(focusTrap: ConfigurableFocusTrap): void {
+    focusTrap.document.removeEventListener('focus', this._listener, true);
+    this._listener = null;
+  }
+
+  /**
+   * Refocuses the first element in the FocusTrap if the focus event target was outside
+   * the FocusTrap.
+   */
+  private _trapFocus(focusTrap: ConfigurableFocusTrap, event: FocusEvent) {
+    const target = event.target as HTMLElement;
+    // Don't refocus if target was in an overlay, because the overlay might be associated
+    // with an element inside the FocusTrap, ex. mat-select.
+    if (!focusTrap.element.contains(target) &&
+      target.closest('div.cdk-overlay-pane') === null) {
+        // Some legacy FocusTrap usages have logic that focuses some element on the page
+        // just before FocusTrap is destroyed. For backwards compatibility, wait
+        // to be sure FocusTrap is still enabled before refocusing.
+        setTimeout(() => {
+          if (focusTrap.enabled) {
+            focusTrap.focusFirstTabbableElement();
+            event.stopImmediatePropagation();
+            event.preventDefault();
+            event.stopPropagation();
```

Comment:
I don't think this will do anything, because it's happening inside a timeout where the event has already bubbled all the way up.

Reply:
This is true. Plus focus events don't bubble anyways, so stopPropagation and stopImmediatePropagation aren't needed. Removed.